### DISCUSSION TITLE
Password Recovery for Appwrite Cloud

### DISCUSSION THREAD

**Discussion by heshamz** [appwrite/appwrite#5447] [Q&A] [answered]
Hi,

I would like to know what would be the URL value for password recovery in the case of Appwrite Cloud.

Thanks

**Comment by loks0n** [comment, marked as answer]
Hi there 👋,

For password recovery, the 'url' parameter is where your user will be redirected after clicking the recovery link.
See the docs [here](https://appwrite.io/docs/client/account?sdk=rest-default#accountCreateRecovery).

For example, 
1. Your app triggers and recovery session for user with email 'myuser@gmail.com' and url 'https://myapp.com/recover'.
2. Appwrite will send an email containing a confirmation link.
3. Once the user clicks the confirmation link, they will be redirected to the url with the secret key and user id values attached to the URL query string. e.g. 'https://myapp.com/recover?userId=13984718947&secret=3918473298471'.
4. Your app may then update the account by reading the values from the query string. See docs [here](https://appwrite.io/docs/client/account#accountUpdateRecovery).

Let me know if you have more questions. 💬 

Best of luck with your project!

**Reply by heshamz** [reply]
Hi,

Why we couldn't use the Appwrite cloud server URL?

**Reply by loks0n** [reply]
Hi 👋,

Great question! The Appwrite Cloud server URL isn't used for password recovery because it's your application that should handle this process, not the server. The password recovery link should lead to a page in your application where users can reset their password.

Keep the questions coming! We're here to help! 💪🚀
